linux_os/guide/services/ntp/chronyd_specify_remote_server/ansible/shared.yml

```diff
@@ -5,18 +5,41 @@
 # disruption = low
 {{{ ansible_instantiate_variables("var_multiple_time_servers") }}}
 
-- name: "Detect if chrony is already configured with pools or servers"
-  find:
-    path: /etc
-    patterns: chrony.conf
-    contains: '^[\s]*(?:server|pool)[\s]+[\w]+'
-  register: chrony_servers
+- name: {{{ rule_title }}} Remove file for remediation
+  ansible.builtin.file:
+    path: ~/set_time_servers.sh
+    state: absent
 
-- name: "Configure remote time servers"
-  lineinfile:
-    path: {{{ chrony_conf_path }}}
-    line: 'server {{ item }}'
-    state: present
-    create: True
-  loop: '{{ var_multiple_time_servers.split(",") }}'
-  when: chrony_servers.matched == 0
+- name: {{{ rule_title }}} Generation of script for remediation
+  ansible.builtin.copy:
```

Comment:
I am not fan of this approach of creating shell scripts in Ansible to be executed by the Playbook itself. Ansible tasks should be accomplished by using specific Ansible modules.

Reply:
Hello @marcusburghardt
I was also trying to use "plain vanilla" ansible - but I did not find a way to implement this logic - to have a loop and inside the loop to have 2 conditions - 1st which checks for existence of a line which has a time server as a sub-string, 2nd which checks if the line begins with the word "server" and  if both conditions are not met to be a added a line to the configuration file which starts with word server, has a space after, and after the space we have the time server. Would be possible to propose me a working example in ansible which covers this logic? It will help me a lot.
All the best Rumen